### DISCUSSION TITLE
Reporting output volume back to controlling device

### DISCUSSION THREAD

**Discussion by fesc2000** [mikebrady/shairport-sync#1472] [General]
Hi,
i have a question regarding the airplay protocol. Some background:

I am feeding my amplifier from an embedded linux runnning three pieces of software:
- shairport-sync
- mpd
- a bluetooth daemon

The volume is actually controlled by another daemon which directly changes and queries the volume as set in the amplifier. So the software always plays at 100%, and the actual "mixer" applies the requested volume to the amplifier.

I have modified the mpd and the bluetooth software so that they are aware of the current amplifier setting. When i change the volume directly on the amplifier, this volume change is reported back all the way to the controlling client (= the mpd player on my phone). Also, if i set the volume via mpd and then switch to bluetooth, the volume level is preserved.

shairport-sync does not really fit in. Whenever i activate it, the volume is getting reset to whatever the player on the iphone thinks it needs to apply, and volume changes on the amplifier are not recognized by the iphone.

So my question would be if this is a limitation of the airport protocol itself, or "just" some missing feature in shairport-sync. If it's the latter i would be happy to integrate it.

**Comment by mikebrady** [comment]
Thanks. It's a suggestion that comes up quite often, but it's really a limitation of AirPlay. Sending information back to the client is tricky and unreliable, I'm afraid.
